phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -242,6 +260,12 @@ public static Expression pushKeyExpressionsToScan(StatementContext context, Filt
                     cnf.add(leftRanges);
                     clipLeftSpan = 0;
                     prevSortOrder = sortOrder;
+                    // since we have to clip the portion with the same sort order, we can no longer
+                    // extract the nodes from the where clause
+                    // for eg. for the schema A VARCHAR DESC, B VARCHAR ASC and query WHERE (A,B) < ('a','b')
+                    // the range (* - a\xFFb) is converted to (~a-*)(*-b)
+                    // so we still need to filter on A,B
```

Comment:
I think the comment should be : we still need to filter on b

Reply:
This needs to filter for both A and B still since the filter on B requires looking at the value of A.

Reply:
So 2 clarifications:
One there is no attempt today to unentwine the RVC comparison into pieces for different columns.
Two, the filter still needs A as the behavior is different if A == 'a' or if A > 'a', if A=='a' then we consider B only, otherwise we do not consider B.  The scan considers rows both A == 'a' and A > 'a'

Reply:
ok cool